From 7f7e118de75a875140fe88187800df4e51670a4c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?V=C3=ADctor=20Manuel=20J=C3=A1quez=20Leal?=
 <vjaquez@igalia.com>
Date: Thu, 16 Mar 2023 15:04:51 +0100
Subject: [PATCH 67/77] video: remove spurious gst_caps_make_writable()

In gst_video_info_dma_drm_to_caps() the caps are newly created, so there's no
need for make it writable. In gst_video_info_dma_drm_from_caps() a copy of the
caps is done, which implies a gst_caps_make_writable().

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/4195>
---
 gst-libs/gst/video/video-info-dma.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/gst-libs/gst/video/video-info-dma.c b/gst-libs/gst/video/video-info-dma.c
index 350f01ea..c712b35b 100644
--- a/gst-libs/gst/video/video-info-dma.c
+++ b/gst-libs/gst/video/video-info-dma.c
@@ -271,8 +271,6 @@ gst_video_info_dma_drm_to_caps (const GstVideoInfoDmaDrm * drm_info)
     return NULL;
   }
 
-  caps = gst_caps_make_writable (caps);
-
   str = gst_video_dma_drm_fourcc_to_string (drm_info->drm_fourcc,
       drm_info->drm_modifier);
 
@@ -318,7 +316,7 @@ gst_video_info_dma_drm_from_caps (GstVideoInfoDmaDrm * drm_info,
 
   GST_DEBUG ("parsing caps %" GST_PTR_FORMAT, caps);
 
-  tmp_caps = gst_caps_make_writable (gst_caps_copy (caps));
+  tmp_caps = gst_caps_copy (caps);
   structure = gst_caps_get_structure (tmp_caps, 0);
 
   str = gst_structure_get_string (structure, "drm-format");
-- 
2.40.1

